fdbserver/VersionedBTree.actor.cpp

```diff
@@ -1286,108 +1289,241 @@ int nextPowerOf2(uint32_t x) {
+				for(auto r = allReasons.begin(); r != allReasons.end(); r++){
+					std::string num = std::to_string(eventReasons[(size_t)e->first][(size_t)r->first]);
+					result += r->second; 
+					result.append(16 - r->second.length(), ' ');
+					result.append(8 - num.length(), ' ');
+					result += num;
+					result.append(13, ' ');
+				}
+				result +="\n";
+			}
+			return result;
+		}
+		void reportTrace(TraceEvent* t, int h){
+			const std::vector<std::pair<events,pagerEventReasons>> possibleEventReasonPairs = {
+				{events::pagerCacheLookup, pagerEventReasons::pointRead},{events::pagerCacheLookup, pagerEventReasons::rangeRead},
+				{events::pagerCacheLookup, pagerEventReasons::rangePrefetch},{events::pagerCacheLookup, pagerEventReasons::commit},
+				{events::pagerCacheLookup, pagerEventReasons::lazyClear},{events::pagerCacheLookup, pagerEventReasons::metaData},
 
+				{events::pagerCacheHit, pagerEventReasons::pointRead},{events::pagerCacheHit, pagerEventReasons::rangeRead},
+				{events::pagerCacheHit, pagerEventReasons::rangePrefetch},
+				{events::pagerCacheHit, pagerEventReasons::lazyClear},{events::pagerCacheHit, pagerEventReasons::metaData},
+
+				{events::pagerCacheMiss, pagerEventReasons::pointRead},{events::pagerCacheMiss, pagerEventReasons::rangeRead},
+				{events::pagerCacheMiss, pagerEventReasons::rangePrefetch},
+				{events::pagerCacheMiss, pagerEventReasons::lazyClear},{events::pagerCacheMiss, pagerEventReasons::metaData},
+
+				{events::pagerWrite, pagerEventReasons::commit},
+				{events::pagerWrite, pagerEventReasons::lazyClear},{events::pagerWrite, pagerEventReasons::metaData},
+			};
+			for ( const auto &ER : possibleEventReasonPairs ){
+				t->detail(format("L%d%s", 
+								 h + 1, 
+								 (eventsCodes[(size_t)ER.first]+pagerEventReasonsCodes[(size_t)ER.second]).c_str()), 
+								 eventReasons[(size_t)ER.first][(size_t)ER.second]
+								);
+			}
+		}
+
+	};
+
+	// Page levle events 
 	struct Level {
-		unsigned int pageRead;
-		unsigned int pageReadExt;
-		unsigned int pageBuild;
-		unsigned int pageBuildExt;
-		unsigned int pageCommitStart;
-		unsigned int pageModify;
-		unsigned int pageModifyExt;
-		unsigned int lazyClearRequeue;
-		unsigned int lazyClearRequeueExt;
-		unsigned int lazyClearFree;
-		unsigned int lazyClearFreeExt;
-		unsigned int forceUpdate;
-		unsigned int detachChild;
-		double buildStoredPct;
-		double buildFillPct;
-		unsigned int buildItemCount;
-		double modifyStoredPct;
-		double modifyFillPct;
-		unsigned int modifyItemCount;
+		struct Metrics{
+			unsigned int pageRead;
+			unsigned int pageReadExt;
+			unsigned int pageBuild;
+			unsigned int pageBuildExt;
+			unsigned int pageCommitStart;
+			unsigned int pageModify;
+			unsigned int pageModifyExt;
+			unsigned int lazyClearRequeue;
+			unsigned int lazyClearRequeueExt;
+			unsigned int lazyClearFree;
+			unsigned int lazyClearFreeExt;
+			unsigned int forceUpdate;
+			unsigned int detachChild;
+			double buildStoredPct;
```

Comment:
These metrics, from `buildStoredPct` down to `modifyItemCount`, currently track totals and item counts useful for calculating an average.  Now that we have histograms for all of these things, I think we can remove them as they are less useful than the histograms.